src/parser/hunter/beastmastery/modules/spells/BarbedShot.tsx

```diff
@@ -59,71 +75,75 @@ class BarbedShot extends Analyzer {
 
   getAverageBarbedShotStacks() {
     let avgStacks = 0;
-    this.barbedShotStacks.forEach((elem, index) => {
-      avgStacks += elem.reduce((a, b) => a + b, 0) / this.owner.fightDuration * index;
+    this.barbedShotStacks.forEach((elem: any[], index: number) => {
+      avgStacks += elem.reduce((a, b) => a + b, 0) /
+        this.owner.fightDuration *
+        index;
     });
     return avgStacks.toFixed(2);
   }
 
-  on_byPlayer_damage(event) {
-    const spellId = event.ability.guid;
-    if (spellId !== SPELLS.BARBED_SHOT.id) {
-      return;
-    }
-    this.damage += event.amount + (event.absorbed || 0);
-  }
-
-  on_toPlayerPet_removebuff(event) {
+  on_toPlayerPet_removebuff(event: RemoveBuffEvent) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.BARBED_SHOT_PET_BUFF.id) {
       return;
     }
     this.handleStacks(event);
   }
 
-  on_byPlayer_applybuff(event) {
+  on_byPlayer_applybuff(event: ApplyBuffEvent) {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.BARBED_SHOT_PET_BUFF.id) {
       return;
     }
     this.handleStacks(event);
   }
 
-  on_byPlayer_applybuffstack(event) {
+  on_byPlayer_applybuffstack(event: ApplyBuffStackEvent): void {
```

Comment:
```suggestion
  on_byPlayer_applybuffstack(event: ApplyBuffStackEvent) {
```

I believe it's best practice to leave as much implicit as possible to make it quicker to iterate.